app/models/concerns/shared/is_data.rb

```diff
@@ -47,31 +108,75 @@ def errors_excepting(*keys)
     errors
   end
 
+  # @return [Scope]
+  def similar
+    klass = self.class
+    attr  = Stripper.new.strip_similar_attributes(klass, attributes)
+    # matching only those attributes from the instance which are not empty
+    attr  = attr.select { |_kee, val| val.present? }
+    if id
+      scope = klass.where(attr).not_self(self)
+    else
+      scope = klass.where(attr)
+    end
+    scope
+  end
+
+  # @return [Scope]
+  def identical
+    klass = self.class
+    attr  = Stripper.new.strip_identical_attributes(klass, attributes)
+    if id
+      scope = klass.where(attr).not_self(self)
+    else
+      scope = klass.where(attr)
+    end
+    scope
+  end
+
+  # @param [Symbol] keys
+  # @return [Array]
   def full_error_messages_excepting(*keys)
     errors_excepting(*keys).full_messages
   end
 
-  module ClassMethods
+  class Stripper
```

Comment:
@TuckerJD just one request- make this a module, not a class (you have no attributes etc. that you need to track across instances).  The invocations will be Stripper.<method> (remove new) and the methods will need the def self.<method>.